gossip/src/crds_gossip_push.rs

```diff
@@ -157,6 +157,12 @@ impl CrdsGossipPush {
                         received_cache.record(origin, from, /*num_dups:*/ usize::MAX);
                         self.num_old.fetch_add(1, Ordering::Relaxed);
                     }
+                    Err(CrdsError::RedundantPull) => {
+                        // rewarded for being after Pull but first Push??
+                        received_cache.record(origin, from, /*num_dups:*/ 0);
+                        // still track that it is old message
+                        self.num_old.fetch_add(1, Ordering::Relaxed);
+                    }
```

Comment:
Question: should the peer that sends the first push message that triggers the discovery of a redundant pull be treated the same in `ReceivedCache` as the peer who sends the first push message resulting in a successful update of the `crds.table`?  I would argue the answer is yes they should be treated the same:
Reasoning: a peer shouldn't be punished just because a `PullRequest` happened to beat it in reporting a datapoint. This also means the second peer would still gets its scored increased in `ReceivedCache` since `NUM_DUPS_THRESHOLD == 2`. Even though the peer may be slow it is still faster than any peer sending this same message after it. 

Possible argument against: Technically the peer wasn't the first one and this is a duplicate. There may be a reason besides chance that a `PullRequest` beat the `Push`.

Reply:
> should the peer that sends the first push message that triggers the discovery of a redundant pull be treated the same in ReceivedCache as the peer who sends the first push message resulting in a successful update of the crds.table?

yes, I think so.

> Possible argument against: Technically the peer wasn't the first one and this is a duplicate. There may be a reason besides chance that a PullRequest beat the Push.

I can't think of a good reason here. Basically if there was no outgoing pull-request this value would have been updated from push anyways.

Reply:
ya i agree